### ISSUE TITLE
Changing opaque type to type alias results in MAJOR bump

### ISSUE THREAD

**Issue by gampleman** [elm/compiler#1895]
```elm
module Foo exposing (Foo)

type Foo =
     Foo Int
```

Gets changed to 

```elm
module Foo exposing (Foo)

import Internal

type alias Foo =
     Internal.Foo

```

```elm
module Internal exposing (Foo)

type Foo =
     Foo Int
```

where `Internal` is *not* in `"exposed-modules"`. Conceptually, this doesn't change anything for the calling code of this package, as it is still the same type, just it's implementation details changed slightly (but being able to do that was the point of making it opaque in the first place). As such, this should be a PATCH change.

However, `elm diff` gives:

```
---- Foo - MAJOR ----

    Added:
        type alias Foo =
            Internal.Foo 

    Removed:
        type Foo
```

- **Elm:** 0.19
- **Operating System:** OSX 10.14.3

**Comment by drathier** [comment]
Changing the custom type to a type alias essentially exposes a constructor to it, so it makes sense the way it is now. Recusing down the types to figure out if it's possible to actually construct that type alias is probably quite a lot of effort.

**Comment by harrysarson** [comment]
> Changing the custom type to a type alias essentially exposes a constructor to it

In which case should this be a MINOR change, equivalent to adding a new function?

**Comment by gampleman** [comment]
I don’t believe this is true: the constructor is not exposed in either scenario.
